pkg/providers/gcp/scheduler.go

```diff
@@ -24,122 +23,107 @@ type gcpScheduler struct {
 	// Project ID
 	project string
 
-	// GCP Metrics client
-	gcpClient *GCP
-
-	// GCE client
-	gce *gceClient
-
-	// Account config
-	account config.Account
+	// GCP handler
+	gcp *GCP
 
 	// Shutdown function
 	shutdown func()
 }
 
-// Return the scheduler interface
-func NewScheduler(eventBus bus.Bus) []v1.Scheduler {
-	// Load the config
-	gcpConfig, exists := config.AppConfig().Providers[gcpProvider]
+// NewScheduler returns the a list of schedulers that
+// conform to the scheduler interface
+func NewScheduler(ctx context.Context, eventBus bus.Bus) []v1.Scheduler {
+	// Load the GCP config
+	cfg, exists := config.AppConfig().Providers[provider]
 
 	// If the provider is not configured - skip its initialization
 	if !exists {
 		return nil
 	}
 
-	// Schedulers for each account
 	var schedulers []v1.Scheduler
 
-	for index := range gcpConfig.Accounts {
-		account := gcpConfig.Accounts[index]
+	// Create a scheduler for each GCP project
+	for index := range cfg.Accounts {
+		account := cfg.Accounts[index]
 
-		// Init the ticket
 		ticker := time.NewTicker(config.AppConfig().ProvidersConfig.Interval)
 
-		// Init the GCE client
-		gce := newGCECLient(&account)
-		if gce == nil {
-			klog.Error("failed to Initialize GCP provider")
-			return nil
-		}
-
-		// Init the GCP metrics Client
-		// This doesn't return metrics, it returns a new GCP client
-		gcpClient, shutdown, err := New(&account, gce.cache)
+		// Init the GCP Client
+		gcp, shutdown, err := New(ctx, &account)
 		if err != nil {
 			klog.Errorf("failed to Initialize GCP provider %s", err)
 			return nil
 		}
 
-		// List all the instances
-		gce.Refresh(account.Project)
-
-		accountScheduler := gcpScheduler{
-			ticker:    ticker,
-			done:      make(chan bool),
-			project:   account.Project,
-			account:   account,
-			eventBus:  eventBus,
-			gcpClient: gcpClient,
-			gce:       gce,
-			shutdown:  shutdown,
-		}
+		// loads all instances into cache
+		gcp.Refresh(ctx, account.Project)
```

Comment:
Won't this still only be called at Scheduler creation? Once when main does the scheduler scraping? So, at what point will new instances be picked up?